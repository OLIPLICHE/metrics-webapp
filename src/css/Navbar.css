.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.5rem;
background-color: var(--color-primary-dark);
}

.navbar-title {
  text-transform: uppercase;
  padding: 1.4rem;
  font-weight: bold;
}

.navbar-icons {
  margin-left: 0.5rem;
}

.fa-arrow-circle-right {
  font-size: 1rem;
}


