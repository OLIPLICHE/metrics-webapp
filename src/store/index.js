import {
  combineReducers,
  createStore,
  applyMiddleware,
} from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../reducers.js';

const reducers = combineReducers({
  root: rootReducer,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
