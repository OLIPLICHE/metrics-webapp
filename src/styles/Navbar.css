.navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem;
    background-color: var(--color-primary-dark);
  }
  
  .navbar-title {
    text-transform: uppercase;
  }
  
  .navbar-icons {
    margin-left: 0.5rem;
  }
  