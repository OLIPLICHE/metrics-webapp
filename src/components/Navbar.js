import { NavLink } from 'react-router-dom';
import '../css/Navbar.css';

const Navbar = () => (
  <header className="navbar">
    <NavLink exact to="/">
      <span className="material-icons-outlined">
        <i className="fal fa-chevron-square-left" />
      </span>
    </NavLink>
    <span className="navbar-title">
      Foreign Exchange Market
    </span>
    <span>
      <span className="material-icons-outlined navbar-icons">
        <i className="fal fa-microphone" />
      </span>
      <span className="material-icons-outlined navbar-icons">
        <i className="fal fa-cog" />
      </span>
    </span>
  </header>
);

export default Navbar;
