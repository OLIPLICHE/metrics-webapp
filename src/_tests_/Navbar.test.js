import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configStore.js';
import Navbar from '../components/Navbar.js';

describe('integration test in navbar', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the navbar', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    const heading = screen.getByText(/FOREIGN EXCHANGE MARKET/i);

    expect(heading).toBeInTheDocument();
  });
});
