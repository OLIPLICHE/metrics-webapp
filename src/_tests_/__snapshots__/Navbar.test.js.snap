// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test in navbar matches the snapshot 1`] = `
<header
  className="navbar"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="material-icons-outlined"
    >
      <i
        className="fal fa-chevron-square-left"
      />
    </span>
  </a>
  <span
    className="navbar-title"
  >
    Foreign Exchange Market
  </span>
  <span>
    <span
      className="material-icons-outlined navbar-icons"
    >
      <i
        className="fal fa-microphone"
      />
    </span>
    <span
      className="material-icons-outlined navbar-icons"
    >
      <i
        className="fal fa-cog"
      />
    </span>
  </span>
</header>
`;
